features/steps/givens/attributes.py

```diff
@@ -11,16 +12,40 @@
 class FirstOrFinal(Enum):
   FIRST = auto()
   FINAL = auto()
-register_type(first_or_final=TypeBuilder.make_enum({"first": FirstOrFinal.FIRST, "final": FirstOrFinal.FINAL }))
 
-@gherkin_ifc.step("{attribute} = {value}")
-def step_impl(context, inst, attribute, value):
+class ComparisonOperator (Enum):
+    EQUAL = auto()
+    NOT_EQUAL = auto()
+
+class SubTypeHandling (Enum):
+    INCLUDE = auto()
+    EXCLUDE = auto()
+
+register_type(include_or_exclude_subtypes=TypeBuilder.make_enum({"including subtypes": SubTypeHandling.INCLUDE, "excluding subtypes": SubTypeHandling.EXCLUDE }))
+register_type(first_or_final=TypeBuilder.make_enum({"first": FirstOrFinal.FIRST, "final": FirstOrFinal.FINAL }))
+register_type(equal_or_not_equal=TypeBuilder.make_enum({
+    "=": ComparisonOperator.EQUAL,
+    "!=": ComparisonOperator.NOT_EQUAL
+}))
+
+def check_entity_type(inst, entity_type, handling: SubTypeHandling):
+    handling_functions = {
+        SubTypeHandling.INCLUDE: lambda inst, entity_type: inst.is_a(entity_type),
```

Comment:
Add a comment above L32 explaining that `INCLUDE` will evaluate to True if inst is a subtype of `entity_type` while the second function for `EXCLUDE` will evaluate to True only for an exact type match.  The code by itself is not obvious enough to be self-documenting.